### PR TITLE
[develop2] test python_requires

### PR DESCRIPTION
**Pull request by memsharded**
Allow ``conan create --python-requires``

Close https://github.com/conan-io/conan/issues/12408

### CODE REVIEW

File: conan/api/subapi/graph.py [hunk 1]
```diff
@@ -49,9 +49,11 @@ def load_root_consumer_conanfile(self, path, profile_host, profile_build,
 
     @api_method
     def load_root_test_conanfile(self, path, tested_reference, profile_host, profile_build,
-                                 update=None, remotes=None, lockfile=None):
+                                 update=None, remotes=None, lockfile=None,
+                                 tested_python_requires=None):
         """ Create and initialize a root node from a test_package/conanfile.py consumer
 
+        :param tested_python_requires:
```
Comment (franramirez688) [inline conan/api/subapi/graph.py]: We're missing the docstring

File: conan/cli/commands/create.py [hunk 2]
```diff
@@ -50,8 +52,12 @@ def create(conan_api, parser, *args):
...
                                   lockfile=lockfile)
     if lockfile:
-        # FIXME: We need to update build_requires too, not only ``requires``
-        lockfile.add(requires=[ref])
+        if args.python_require:
+            lockfile.add(python_requires=[ref])
+        elif args.build_require:
+            lockfile.add(build_requires=[ref])
+        else:
+            lockfile.add(requires=[ref])
```
Comment (franramirez688) [inline conan/cli/commands/create.py]: Should we add a test to check this?

**Review by franramirez688 — COMMENTED** [review]
I think it looks good!

File: conan/cli/commands/create.py (hunk 2 above)
Comment (memsharded) [reply conan/cli/commands/create.py]: Yes, I'll do it

Comment (memsharded) [reply conan/cli/commands/create.py]: Removed this thing from this PR, will do another PR with more stuff related to lockfiles there are things missing there

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**